tests/at_functional_test/test/enroll_verb_test.dart

```diff
@@ -1404,4 +1404,206 @@ void main() {
+
+      // create a new cram authenticated connection
+      OutboundConnectionFactory secondAuthenticatedConnection =
+          OutboundConnectionFactory();
+      await secondAuthenticatedConnection.initiateConnectionWithListener(
+          firstAtSign, firstAtSignHost, firstAtSignPort);
+      await secondAuthenticatedConnection.authenticateConnection(
+          authType: AuthType.cram);
+
+      // approve then revoke the above created enrollment
+      String approveCommand =
+          'enroll:approve:{"enrollmentId":"$enrollmentId","encryptedDefaultEncryptionPrivateKey":"${apkamEncryptedKeysMap['encryptedDefaultEncryptionPrivateKey']}","encryptedDefaultSelfEncryptionKey":"${apkamEncryptedKeysMap['encryptedSelfEncKey']}","apkamPublicKey":"${at_demos.pkamPublicKeyMap[firstAtSign]!}","encryptedAPKAMSymmetricKey":"${apkamEncryptedKeysMap['encryptedAPKAMSymmetricKey']}"}';
+      enrollmentResponse = await secondAuthenticatedConnection
+          .sendRequestToServer(approveCommand);
+      String revokeCommand = 'enroll:revoke:{"enrollmentId":"$enrollmentId"}';
+      enrollmentResponse = await secondAuthenticatedConnection
+          .sendRequestToServer(revokeCommand);
+      jsonDecodedResponse =
+          jsonDecode(enrollmentResponse.replaceFirst('data:', ""));
+      expect(jsonDecodedResponse['status'], 'revoked');
+
+      // now try to delete the revoked enrollment
+      String deleteEnrollmentCommand =
+          'enroll:delete:{"enrollmentId":"$enrollmentId"}';
+      String revokeEnrollmentResponse = await secondAuthenticatedConnection
+          .sendRequestToServer(deleteEnrollmentCommand);
+      revokeEnrollmentResponse =
+          revokeEnrollmentResponse.replaceFirst('error:', '');
+      jsonDecodedResponse = jsonDecode(revokeEnrollmentResponse);
+      expect(jsonDecodedResponse['errorCode'], 'AT0011');
+      expect(
+          jsonDecodedResponse['errorDescription'],
+          'Internal server exception : Failed to delete enrollment id: '
+          '$enrollmentId | Cause: Cannot delete revoked enrollments. '
+          'Only denied enrollments can be deleted');
+    });
+
+    test(
+        'A test to verify delete operation cannot be performed on an unauthenticated connection',
+        () async {
+      // using a dummy enrollId as the enrollId will NOT be validated for this test
+      String enrollmentId = '023409340293-2342397-23423984729';
+      OutboundConnectionFactory socketConnection2 =
+          await OutboundConnectionFactory().initiateConnectionWithListener(
+              firstAtSign, firstAtSignHost, firstAtSignPort);
+      String deleteEnrollmentCommand =
+          'enroll:delete:{"enrollmentId":"$enrollmentId"}';
+      String revokeEnrollmentResponse =
+          await socketConnection2.sendRequestToServer(deleteEnrollmentCommand);
+      expect(revokeEnrollmentResponse.trim(),
+          'error:AT0401-Exception: Cannot delete enrollment without authentication');
+    });
+  }, skip: true);
```

Comment:
why skipped?

Reply:
I assumed the functional tests would be run against the published docker image which would not contain the current changes. I checked the actions file now and the secondary image is being built using the at_server code in the current branch. The tests have been skipped in Changed made in [3ca6f91](https://github.com/atsign-foundation/at_server/pull/2095/commits/3ca6f9125f3bb4abbf71a69232624c730f6460b6).